core/src/test/java/com/yahoo/imapnio/async/data/SearchResultTest.java

```diff
@@ -16,21 +16,35 @@
      */
     @Test
     public void testSearchResult() {
-        final List<Long> ll = new ArrayList<>();
-        ll.add(Long.MAX_VALUE - 1);
-        final SearchResult infos = new SearchResult(ll);
+        final List<Long> ll = Collections.singletonList(Long.MAX_VALUE - 1);
+        final SearchResult infos = new SearchResult(ll, 1L);
         final List<Long> result = infos.getMessageNumbers();
+        final long modSeq = infos.getHighestModSeq();
```

Comment:
getHighestModSeq returns a Long object, which can be nullable. So get the result in a Long and check for null before check for actual value.